src/util/lineItems.js

```diff
@@ -0,0 +1,168 @@
+  return new Money(
+    convertUnitToSubUnit(numericTotalPrice, unitDivisor(unitPrice.currency)),
+    unitPrice.currency
+  );
+};
+
+const calculateTotalPriceFromSeats = (unitPrice, unitCount, seats) => {
+  const numericPrice = convertMoneyToNumber(unitPrice);
+  const numericTotalPrice = new Decimal(numericPrice)
+    .times(unitCount)
+    .times(seats)
+    .toNumber();
+  return new Money(
+    convertUnitToSubUnit(numericTotalPrice, unitDivisor(unitPrice.currency)),
+    unitPrice.currency
+  );
+};
+
+export const validLineItem = params => {
+  const { code, unitPrice, quantity, percentage, seats, units, includeFor } = params;
+
+  let pricingParams;
+
+  if (quantity) {
+    pricingParams = { quantity, lineTotal: calculateTotalPriceFromQuantity(unitPrice, quantity) };
+  } else if (percentage) {
+    pricingParams = {
+      percentage,
+      lineTotal: calculateTotalPriceFromPercentage(unitPrice, percentage),
+    };
+  } else if (seats && units) {
+    pricingParams = {
+      seats,
+      units,
+      lineTotal: calculateTotalPriceFromSeats(unitPrice, units, seats),
+    };
+  } else {
+    console.error(
+      "Can't calculate the lineTotal of lineItem: ",
+      code,
+      ' Make sure you have provided quantity, percentage or both seats and units'
+    );
+  }
+
+  return {
+    code: code,
+    unitPrice,
+    includeFor: includeFor ? includeFor : ['customer', 'provider'],
+    ...pricingParams,
+  };
+};
+
+export const calculateTotalFromLineItems = lineItems => {
+  const numericTotalPrice = lineItems.reduce((sum, lineItem) => {
+    const numericPrice = convertMoneyToNumber(lineItem.lineTotal);
+    return new Decimal(numericPrice).add(sum);
+  }, 0);
+
+  const unitPrice = lineItems[0].unitPrice;
+
+  return new Money(
+    convertUnitToSubUnit(numericTotalPrice.toNumber(), unitDivisor(unitPrice.currency)),
+    unitPrice.currency
+  );
+};
+
+export const calculateTotalForProvider = lineItems => {
+  const providerLineItems = lineItems.filter(lineItem => lineItem.includeFor.includes('provider'));
+  return calculateTotalFromLineItems(providerLineItems);
+};
+
+export const calculateTotalForCustomer = lineItems => {
+  const providerLineItems = lineItems.filter(lineItem => lineItem.includeFor.includes('customer'));
+  return calculateTotalFromLineItems(providerLineItems);
+};
+
+export const constructLineItems = (startDate, endDate, unitPrice) => {
+  const bookingLineItem = validLineItem({
+    code: 'line-item/nights',
+    unitPrice,
+    quantity: nightsBetween(startDate, endDate),
+    includeFor: ['customer', 'provider'],
+  });
+
+  const cleaningFee = validLineItem({
+    code: 'line-item/cleaning-fee',
+    unitPrice: new Money(7500, 'USD'),
+    quantity: 1,
+    includeFor: ['customer', 'provider'],
+  });
+
+  const providerCommission = validLineItem({
+    code: 'line-item/provider-commission',
+    unitPrice: calculateTotalFromLineItems([bookingLineItem, cleaningFee]),
+    percentage: -15,
```

Comment:
These commission percentages could be defined as variables within this file. That way they are easily customized without looking at the line item generation, and we can also give good documentation on the possible values.